### PR TITLE
LRQA-73907 | master

### PR DESCRIPTION
**Pull request by lilylcl**
https://issues.liferay.com/browse/LRQA-73907

### CODE REVIEW

**Comment by lilylcl** [comment]
ci:test:sf

**Comment by Sharryxue** [comment]
Just starting review.

File: modules/apps/portal-language/portal-language-override-test/src/testFunctional/paths/LanguageOverride.path [hunk 1]
```diff
@@ -69,6 +69,24 @@
 	<td></td>
 	<td></td>
 </tr>
+
+<!--IMPORT_TRANSLATIONS-->
```
Comment (Sharryxue) [inline modules/apps/portal-language/portal-language-override-test/src/testFunctional/paths/LanguageOverride.path:73]: Please double check the new added locator, use the exist one.

File: modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase [hunk 2]
```diff
@@ -21,6 +21,40 @@ definition {
...
+
+			AssertElementPresent(locator1 = "Button#SAVE");
+		}
+
+		task ("When click Cancel button") {
+			Button.clickCancel();
+		}
+
+		task ("Then user is redirected to language key page") {
+			ControlMenu.viewHeaderTitle(headerTitle = "Language Override");
```
Comment (Sharryxue) [inline modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase:58]: Add a step to assert Import Translations header not displays. Thanks

File: modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase [hunk 3]
```diff
@@ -21,6 +21,40 @@ definition {
 		}
 	}
 
+	@description = "This test asserts that Cancel button works as expected in Import Translations page."
+	@priority = "5"
+	test CancelButtonWorks {
```
Comment (Sharryxue) [inline modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase]: Shall we rename the test to CanCancelImportTranslation. Thanks

File: modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase [hunk 4]
```diff
@@ -21,6 +21,40 @@ definition {
...
+				locator1 = "LanguageOverride#IMPORT_TRANSLATIONS_DESCRIPTION",
+				value1 = "Support file format .properties.");
+
+			AssertElementPresent(
+				defaultLanguage = "en-us",
+				locator1 = "Select#DEFAULT_LANGUAGE");
+
+			AssertElementPresent(locator1 = "TextInput#FILE");
+
+			AssertElementPresent(locator1 = "Button#SAVE");
```
Comment (Sharryxue) [inline modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase]: Find the exist macro first.

**Review by Sharryxue — CHANGES_REQUESTED** [review]
Hello @lilylcl,
Please remember to use the exist locator and macro first, thanks.

File: modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase (hunk 3 above)
Comment (lilylcl) [reply modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase]: Done.

File: modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase (hunk 2 above)
Comment (lilylcl) [reply modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase:58]: Done.

File: modules/apps/portal-language/portal-language-override-test/src/testFunctional/paths/LanguageOverride.path (hunk 1 above)
Comment (lilylcl) [reply modules/apps/portal-language/portal-language-override-test/src/testFunctional/paths/LanguageOverride.path:73]: No exist, needs to add and keep it

File: modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase (hunk 4 above)
Comment (lilylcl) [reply modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase]: No exist, needs to add and put in LanguagePortlet.macro

**Comment by lilylcl** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/macros/sitenavigation/languageportlet/LanguagePortlet.macro [hunk 5]
```diff
@@ -91,6 +91,24 @@ definition {
...
+	macro viewImportTranslations {
+		AssertTextEquals(
+			locator1 = "LanguageOverride#IMPORT_TRANSLATIONS_TITLE",
+			value1 = "${heading}");
+
+		AssertTextEquals(
+			locator1 = "LanguageOverride#IMPORT_TRANSLATIONS_DESCRIPTION",
+			value1 = "${description}");
+
+		AssertElementPresent(
```
Comment (Sharryxue) [inline portal-web/test/functional/com/liferay/portalweb/macros/sitenavigation/languageportlet/LanguagePortlet.macro]: use exist macro.

File: modules/apps/portal-language/portal-language-override-test/src/testFunctional/paths/LanguageOverride.path [hunk 6]
```diff
@@ -69,6 +69,24 @@
...
+
+<tr>
+	<td>IMPORT_TRANSLATIONS_TITLE</td>
```
Comment (Sharryxue) [inline modules/apps/portal-language/portal-language-override-test/src/testFunctional/paths/LanguageOverride.path]: Move it to a more general path file, and write in detail.

File: portal-web/test/functional/com/liferay/portalweb/macros/sitenavigation/languageportlet/LanguagePortlet.macro [hunk 7]
```diff
@@ -91,6 +91,24 @@ definition {
...
+			key_locale = "${locale}",
+			locator1 = "Select#DEFAULT_LANGUAGE");
+
+		AssertElementPresent(locator1 = "TextInput#FILE");
+
+		AssertElementPresent(locator1 = "Button#SAVE");
```
Comment (Sharryxue) [inline portal-web/test/functional/com/liferay/portalweb/macros/sitenavigation/languageportlet/LanguagePortlet.macro]: Use the exist macro.

File: portal-web/test/functional/com/liferay/portalweb/macros/sitenavigation/languageportlet/LanguagePortlet.macro [hunk 8]
```diff
@@ -91,6 +91,24 @@ definition {
...
+
+		AssertTextEquals(
+			locator1 = "LanguageOverride#IMPORT_TRANSLATIONS_DESCRIPTION",
+			value1 = "${description}");
+
+		AssertElementPresent(
+			key_locale = "${locale}",
+			locator1 = "Select#DEFAULT_LANGUAGE");
+
+		AssertElementPresent(locator1 = "TextInput#FILE");
```
Comment (Sharryxue) [inline portal-web/test/functional/com/liferay/portalweb/macros/sitenavigation/languageportlet/LanguagePortlet.macro]: Use the exist macro.

**Review by Sharryxue — CHANGES_REQUESTED** [review]
Hello @lilylcl,
Please try to find the exist macro and use them thanks.

**Comment by lilylcl** [comment]
ci:test:sf

**Comment by Sharryxue** [comment]
Just starting review.

**Comment by Sharryxue** [comment]
Hello @lilylcl,
Please update the test based on the comment, thanks.

File: modules/apps/portal-language/portal-language-override-test/src/testFunctional/paths/LanguageOverride.path (hunk 6 above)
Comment (lilylcl) [reply modules/apps/portal-language/portal-language-override-test/src/testFunctional/paths/LanguageOverride.path]: Done.

**Comment by lilylcl** [comment]
Resend [SShi#474](https://github.com/Sharryxue/liferay-portal/pull/474)

**Pull request closed without merging**